### PR TITLE
SERVICES-989 marketplace events indexing admin endpoints

### PR DESCRIPTION
**Pull request by johnykes**

### CODE REVIEW

File: src/modules/admins/admin-operations.resolver.ts [hunk 1]
```diff
@@ -93,4 +95,29 @@ export class AdminOperationsResolver {
       throw new ApolloError(error);
     }
   }
+
+  @Mutation(() => Boolean)
+  @UseGuards(GqlAdminAuthGuard)
```
Comment (danielailie) [inline src/modules/admins/admin-operations.resolver.ts:100]: could you extract in a args object all this parameters

**Review by danielailie — CHANGES_REQUESTED** [review]

File: src/modules/admins/admin-operations.resolver.ts (hunk 1 above)
Comment (johnykes) [reply src/modules/admins/admin-operations.resolver.ts:101]: Done

File: src/modules/marketplaces/models/MarketplaceEventsIndexingInput.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { MarketplaceEventsIndexingArgs } from './MarketplaceEventsIndexingArgs';
+
```
Comment (danielailie) [inline src/modules/marketplaces/models/MarketplaceEventsIndexingInput.ts:2]: can you change it to MarketplaceEventsIndexingRequest just to be same name as the others

Comment (johnykes) [reply src/modules/marketplaces/models/MarketplaceEventsIndexingInput.ts]: Done

**Review by danielailie — APPROVED** [review]

File: src/modules/marketplaces/marketplaces-events-indexing.service.ts [hunk 3]
```diff
@@ -25,6 +27,33 @@ export class MarketplaceEventsIndexingService {
...
+    beforeTimestamp?: number,
+    afterTimestamp?: number,
+    stopIfDuplicates?: boolean,
+  ): Promise<void> {
+    await Locker.lock(
+      'reindexAllMarketplaceEvents',
+      async () => {
+        let [marketplaces] = await this.persistenceService.getMarketplaces();
+        let marketplaceAddresses = [
+          ...new Set(marketplaces.map((marketplace) => marketplace.address)),
```
Comment (gabrielmatei) [inline src/modules/marketplaces/marketplaces-events-indexing.service.ts:40]: maybe you can use `.distinct()` prototype from erdnest extendions in the future

**Review by gabrielmatei — APPROVED** [review]

**Pull request merged**